src/raven/client.clj

```diff
@@ -385,6 +387,38 @@
     :data data
     :env env}))
 
+(defn get-full-ring-url
+  "Given a ring compliant request object, return the full URL.
+    This was lifted from ring's source code so as to not depend on it."
+  [request]
+  (str (-> request :scheme name)
+       "://"
+       (get-in request [:headers "host"])
+       (:uri request)
+       (if-let [query (:query-string request)]
```

Comment:
when-let ?

Reply:
Done, thanks.

I also added a new test covering this case, to make sure I didn't break anything.

PS: This should be brought up here as well: https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/util/request.clj#L13